### PR TITLE
[ASTS] Foreground Task II.B: Add mock-based tests for DefaultSingleBucketSweepTask

### PR DESCRIPTION
**Pull request by jeremyk-91**
## General
**Before this PR**: #7270 doesn't have tests. Some were proposed in #7288, but those are styled more in the way of integration tests - while they provide value in terms of coverage, identifying what exactly went wrong for those is not exactly trivial.

**After this PR**:

==COMMIT_MSG==
`DefaultSingleBucketSweepTask` has mock-based tests.
==COMMIT_MSG==

**Priority**: P2

**Concerns / possible downsides (what feedback would you like?)**: Are there any missing test cases? Otherwise nothing in particular.

**Is documentation needed?**: No.

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**: No

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**: No

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**: Yes

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**: No

**Does this PR need a schema migration?** No

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**: There are some assumptions about how a sweep queue reader etc. implicitly work in terms of the stubbing. (The integration tests should, however, catch this kind of thing.)

**What was existing testing like? What have you done to improve it?**: This PR only adds tests.

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**: N/A

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**: N/A

## Execution
This PR only modifies tests.

## Scale
This PR only modifies tests.

## Development Process
**Where should we start reviewing?**: The one file

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+                                DedicatedRows.of(ImmutableList.of()),
+                                context.endTimestampExclusive() - 1),
+                        ImmutableSet.of()));
+        when(boundRetriever.getStrictUpperBoundForCompletelyClosedBuckets()).thenReturn(Long.MAX_VALUE);
+
+        defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket());
+        verify(sweepQueueDeleter).sweep(eq(ImmutableList.of()), eq(Sweeper.of(context.shardAndStrategy())));
+        verify(sweepQueueCleaner)
+                .clean(
+                        eq(context.shardAndStrategy()),
```
Comment (mdaudali) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java]: nits: You don't need eq() around the concrete objects if you're applying it to all arguments! Was there a readability preference for including it?

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+                .updateBucketProgressToAtLeast(
+                        context.bucket(),
+                        BucketProgress.createForTimestampProgress(
+                                sweepTimestamp - 1 - context.startTimestampInclusive()));
+        verifyNoInteractions(completionListener);
+    }
+
+    @ParameterizedTest
+    @MethodSource("allSweepBuckets")
+    public void doesNotRemoveSweepDataOutsideBucket(SweepBucketTestContext context) {
```
Comment (mdaudali) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java:262]: note to self to check tomorrow: Is this just checking that we don't also remove the fine partition right before startTimestampInclusive?

**Review by mdaudali — COMMENTED** [review]
Makes sense. One nit, and something I want to check tomorrow for test clarity

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java (hunk 2 above)
Comment (jeremyk-91) [reply atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java:262]: Yes, this test was added after I noticed that we need to handle that case correctly!

**Review by mdaudali — APPROVED** [review]

**Pull request merged**